### PR TITLE
[LPT] Fix not thread safe function setDefaultPrecisions

### PR DESCRIPTION
**Pull request by ndemashov**
### Details:
LPT has defaultPrecisions global non-const variable

### Tickets:
55743

### CODE REVIEW

File: src/common/low_precision_transformations/include/low_precision/rt_info/precisions_attribute.hpp [hunk 1]
```diff
@@ -32,6 +32,7 @@ using PrecisionsAttributePtr = std::shared_ptr<PrecisionsAttribute>;
 class LP_TRANSFORMATIONS_API PrecisionsAttribute : public SharedValueAttribute<PrecisionsSharedValue> {
 public:
     static std::vector<ngraph::element::Type> defaultPrecisions;
+    static std::mutex defaultPrecisions_guard;
```
Comment (eshoguli) [inline src/common/low_precision_transformations/include/low_precision/rt_info/precisions_attribute.hpp:35]: 1. please, rename: `defaultPrecisions_guard` => `defaultPrecisionsMutex`
2. make it private

Comment (ndemashov) [reply src/common/low_precision_transformations/include/low_precision/rt_info/precisions_attribute.hpp]: Done, thank you

File: src/common/low_precision_transformations/include/low_precision/rt_info/precisions_attribute.hpp [hunk 2]
```diff
@@ -31,8 +31,7 @@ using PrecisionsAttributePtr = std::shared_ptr<PrecisionsAttribute>;
 
 class LP_TRANSFORMATIONS_API PrecisionsAttribute : public SharedValueAttribute<PrecisionsSharedValue> {
 public:
-    static std::vector<ngraph::element::Type> defaultPrecisions;
-    PrecisionsAttribute(const std::vector<ngraph::element::Type>& precisions = defaultPrecisions);
+    PrecisionsAttribute(const std::vector<ngraph::element::Type>& precisions = { ngraph::element::u8,  ngraph::element::i8 });
```
Comment (eshoguli) [inline src/common/low_precision_transformations/include/low_precision/rt_info/precisions_attribute.hpp:34]: Please, remove default and pass from `LayerTransformation::getDefaultPrecisions()`

Comment (ndemashov) [reply src/common/low_precision_transformations/include/low_precision/rt_info/precisions_attribute.hpp]: Fix it, thank you.

File: src/common/low_precision_transformations/include/low_precision/rt_info/precisions_attribute.hpp [hunk 3]
```diff
@@ -31,8 +31,7 @@ using PrecisionsAttributePtr = std::shared_ptr<PrecisionsAttribute>;
 
 class LP_TRANSFORMATIONS_API PrecisionsAttribute : public SharedValueAttribute<PrecisionsSharedValue> {
 public:
-    static std::vector<ngraph::element::Type> defaultPrecisions;
-    PrecisionsAttribute(const std::vector<ngraph::element::Type>& precisions = defaultPrecisions);
+    PrecisionsAttribute(const std::vector<ngraph::element::Type>& precisions = ngraph::pass::low_precision::LayerTransformation::getDefaultPrecisions());
```
Comment (eshoguli) [inline src/common/low_precision_transformations/include/low_precision/rt_info/precisions_attribute.hpp:34]: remove default argument and dependency on `layer_transformation.hpp` (`#include "low_precision/layer_transformation.hpp"`), please

Comment (eshoguli) [inline src/common/low_precision_transformations/include/low_precision/rt_info/precisions_attribute.hpp:34]: remove default argument and dependency on `layer_transformation.hpp` (`#include "low_precision/layer_transformation.hpp"`), please

Comment (ndemashov) [reply src/common/low_precision_transformations/include/low_precision/rt_info/precisions_attribute.hpp]: moved it to "create" function, thank you

File: src/common/low_precision_transformations/src/rt_info/precisions_attribute.cpp [hunk 4]
```diff
@@ -27,7 +25,8 @@ ov::Any PrecisionsAttribute::create(
     const std::shared_ptr<ngraph::Node>& node,
     const AttributeParameters& params) {
     auto& rt = ov::is_type<opset1::FakeQuantize>(node) ? node->output(0).get_rt_info() : node->get_rt_info();
-    return (rt[PrecisionsAttribute::get_type_info_static()] = PrecisionsAttribute());
+    return (rt[PrecisionsAttribute::get_type_info_static()] = PrecisionsAttribute(
+        ngraph::pass::low_precision::LayerTransformation::getDefaultPrecisions()));
```
Comment (eshoguli) [inline src/common/low_precision_transformations/src/rt_info/precisions_attribute.cpp:29]: It's not good idea to execute `ngraph::pass::low_precision::LayerTransformation::getDefaultPrecisions()` in attribute. To pass precisions as input argument is better. Will be fixed later.

**Review by eshoguli — APPROVED** [review]

File: src/common/low_precision_transformations/src/layer_transformation.cpp [hunk 5]
```diff
@@ -442,6 +446,16 @@ void LayerTransformation::addPattern(ngraph::pass::GraphRewrite& pass, Transform
     NGRAPH_SUPPRESS_DEPRECATED_END
 }
 
+void LayerTransformation::setDefaultPrecisions(const std::vector<ngraph::element::Type>& precisions) {
+    std::lock_guard<std::mutex> lock(defaultPrecisionsMutex);
+    defaultPrecisions = precisions;
```
Comment (ilya-lavrenov) [inline src/common/low_precision_transformations/src/layer_transformation.cpp:451]: how this solved the issue? 
If CPU set one set of precisions and runs LPT pipeline, GPU sets different set of precisions while CPU LPT pipeline is running, then CPU LPT pipeline can be affected anyway

**Pull request merged**